parser.js

```diff
@@ -10,37 +10,35 @@ function Parser() {
     return new Parser()
   }
 
-  this._list = bl()
-  this._newPacket()
-
   this._states = [
       '_parseHeader'
     , '_parseLength'
     , '_parsePayload'
     , '_newPacket'
   ]
-  this._stateCounter = 0
+
+  this._resetState()
 }
 
 inherits(Parser, EE)
 
-Parser.prototype._newPacket = function () {
-  if (this.packet) {
-    this._list.consume(this.packet.length)
-    this.emit('packet', this.packet)
-  }
-
+Parser.prototype._resetState = function () {
   this.packet = new Packet()
-
-  return true
+  this.error = null
+  this._list = bl()
+  this._stateCounter = 0
 }
 
 Parser.prototype.parse = function (buf) {
+  if (this.error) {
+    this._resetState()
+  }
```

Comment:
I could go either way. `_resetState()` resets the error as well as the buffer that had the error in it so it shouldn't keep resetting.

If we don't reset the state, calling `parse()` again with a new buffer is guaranteed to keep emitting the same error over and over since it appends the new buffer after the old.

So we could just tell the user to throw that instance of `Parser` away, or reset the internal state like I'm doing here. If we do choose not to reset for them, should we at least throw an exception to make it obvious the current instance is broken?